lib/server.js

```diff
@@ -84,9 +85,21 @@ function createServer (options, httpHandler) {
 
         multipleBindings.call(this, server, httpHandler, options, listenOptions, () => {
           this[kState].listening = true
+          onListenHookRunner(this)
           cb(null, address)
         })
       }
+    } else {
+      listenOptions.cb = (err, address) => {
+        // the server did not start
+        if (err) {
+          cb(err, address)
+          return
+        }
+        this[kState].listening = true
+        onListenHookRunner(this)
+        cb(null, address)
```

Comment:
Should we return the control faster to the application instead of waiting for all hooks to run? Asking, if we are aiming to not block the app for listening, The best is to return the control asap so we do not block the flow.

Anyways we are logging the errors, so in theory shouldn't be a problem

Reply:
You mean call cb before onListenHookRunner?

Reply:
Yes, exactly 👍